build-llvm.py

```diff
@@ -0,0 +1,402 @@
+                        down an older bug to properly bisect. This value is just passed along to 'git checkout'
+                        so it can be a branch name, tag name, or hash.
+
+                        """),
+                        type=str,
+                        default="master")
+    parser.add_argument("-d",
+                        "--debug",
+                        help=textwrap.dedent("""\
+                        By default, the script builds LLVM in the release configuration with all of
+                        the tests turned off and optimization at O2. This disables that optimization,
+                        builds the tests, and changes the configuration to debug. This can help with
+                        reporting problems to LLVM developers but will make compilation of both LLVM
+                        and the kernel go slower.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-i",
+                        "--incremental",
+                        help=textwrap.dedent("""\
+                        By default, the script removes all build artifacts from previous compiles. This
+                        prevents that, allowing for dirty builds and faster compiles.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-I",
+                        "--install-folder",
+                        help=textwrap.dedent("""\
+                        By default, the script will create a "usr" folder in the same folder as this script
+                        and install the LLVM toolchain there. If you'd like to have it installed somewhere
+                        else, pass it to this parameter. This can either be an absolute or relative path.
+
+                        """),
+                        type=str,
+                        default=os.path.join(root.as_posix(), "usr"))
+    parser.add_argument("-n",
+                        "--no-pull",
+                        help=textwrap.dedent("""\
+                        By default, the script always updates the LLVM repo before building. This prevents
+                        that, which can be helpful during something like bisecting.
+
+                        """),
+                        action="store_true")
+    parser.add_argument("-p",
+                        "--projects",
+                        help=textwrap.dedent("""\
+                        Currently, the script only enables the clang, compiler-rt, and lld folders in LLVM. If
+                        you would like to override this, you can use this parameter and supply a list that is
+                        supported by LLVM_ENABLE_PROJECTS.
+
+                        See step #5 here: https://llvm.org/docs/GettingStarted.html#getting-started-quickly-a-summary
+
+                        Example: -p \"clang;lld;libcxx\"
```

Comment:
I hate this string; if we could make something nicer, I would be so happy. But maybe a TODO for v2.

Reply:
Would you like me to just remove it? I assume if the user wants to actually change it, they can read the documentation for the proper string to supply.

Reply:
No.